feat(e2e): add cypress tests covering most Task Library functionality [SCHOOL-190]

BillClinton opened the pull request.

BillClinton committed:
feat: add a test to verify button disabled state

BillClinton committed:
feat: add a test to verify button disabled state

BillClinton committed:
fix: resolve merge conflicts after rebase

themightychris commented on cypress/integration/cbl/admin/tasks-manager.js:
```diff
@@ -178,6 +178,39 @@ describe('CBL / Admin / Tasks Manager', () => {
+            .should('have.class', 'x-btn-disabled')
+
+        // select the first grid cell
+        cy.root()
+            .get('.x-grid-cell-inner')
```
In this case, using `cy.root()` is redundant:

```suggestion
        cy.get('.x-grid-cell-inner')
```

BillClinton replied:
resolved [3a98c74](https://github.com/SlateFoundation/slate-cbl/pull/732/commits/3a98c74dd03dbe6899a345a07d1ee26c56d779a9)

BillClinton committed:
feat: add tests for sorting by column

BillClinton committed:
feat: make assertion text more specific

BillClinton committed:
feat: add archived task for archive filter testing

BillClinton committed:
fix: rework status conditions to address bug

In creating cypress tests for the status filters, conflicts were observed.
Specifically, archived tasks would not be displayed unless unshared tasks
were also requested.  This fix combines the creation of conditions for
unshared and archived tasks into one function.

BillClinton committed:
refactor: reduce _fn use for increased clarity

BillClinton committed:
fix: correct typo in assertion text

BillClinton committed:
fix: set up test properly

BillClinton committed:
feat: add test to verify archive filter

BillClinton committed:
fix: remove unnecessary call to cy.root

themightychris committed:
fix(e2e): uncomment resetDatabase command

themightychris committed:
feat(e2e): add assertion on edit field content

themightychris reviewed (approved):
@BillClinton you did an _incredible_ job on this! Perfectly executed in nearly every way

One mistake I caught was that you left `resetDatabase()` commented out. I also added one assertion on the existing field value during the edit test just for completeness
